wikimedia/wikipedia-ios
# Improve iOS 14 deep-press back button behavior

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T262305

### Notes
Three things in this PR:
1. Like in normal navigation, remove the SearchController from the navigation stack once you've navigated more than one VC beyond the SearchController.
2. On the SearchController, improve the code a bit. No behavior changes.
3. Comment updates, for accuracy.

### Test Steps
1. On an iOS 14 device/simulator.
1. Open a few controllers. Deep press the back button. Ensure you see a list of all VCs you've tapped through, and can jump back to any of them.
1. Check for regressions: Open a Search Controller. Search, tap into an article*, and tap into another article. Tap back button twice, ensure you never see the Search Controller.
1. Confirm new code is good: Open a Search Controller. Search, tap into an article*, and deep press the back button. Ensure the first item is `Search`. Tap it, ensure you jump back to SearchController.
1. Confirm new code is good (con't): Search, tap into an article*, tap into another article, and deep press the back button. Ensure the `Search` item isn't visible in the list. Tap into a prior article, make sure you jump back to it as expected.

* On this screen, instead of `< Back` you should see `< Search`

### Screenshots/Videos
<img width="200" alt="Screenshot 79" src="https://user-images.githubusercontent.com/9295855/93138390-ca20e400-f693-11ea-952c-ff0a433e7c9f.png">
